src/vcpkg-test/configparser.cpp

```diff
@@ -40,23 +60,35 @@ TEST_CASE ("BinaryConfigParser files provider", "[binaryconfigparser]")
     }
     {
         auto parsed = create_binary_providers_from_configs_pure("files," ABSOLUTE_PATH, {});
-        REQUIRE(parsed.has_value());
+        auto state = parsed.value_or_exit(VCPKG_LINE_INFO);
+
+        REQUIRE(state.binary_cache_providers == std::set<StringLiteral>{{"default"}, {"files"}});
     }
     {
         auto parsed = create_binary_providers_from_configs_pure("files," ABSOLUTE_PATH ",nonsense", {});
         REQUIRE(!parsed.has_value());
     }
     {
         auto parsed = create_binary_providers_from_configs_pure("files," ABSOLUTE_PATH ",read", {});
-        REQUIRE(parsed.has_value());
+        auto state = parsed.value_or_exit(VCPKG_LINE_INFO);
+
+        REQUIRE(state.binary_cache_providers == std::set<StringLiteral>{{"default"}, {"files"}});
+        REQUIRE(!state.archives_to_read.empty());
```

Comment:
```suggestion
        REQUIRE(!state.archives_to_read.empty());
        REQUIRE(state.archives_to_write.empty());
```

should we add this test?